src/main.rs

```diff
@@ -178,24 +238,37 @@ fn main() {
     .collect();
 
     let args = Args::parse();
+
+    // Establish the architecture we're dealing with and arch-specific mode setting
+    let arch = args.get_arch();
+
+    // Read the file
     let f = File::open(args.path).expect("can't open object file");
     let buf = unsafe { memmap::Mmap::map(&f).expect("can't memmap object file") };
-
     let obj = object::File::parse(&*buf).expect("can't parse object file");
 
-    let mode = if obj.is_64() {
-        Mode::Mode64
-    } else {
-        Mode::Mode32
+    // The file open mode is quite processor-specific and for x86 is determined by the file we just opened.
+    let (mode, describe_group): (Mode, fn(&u8) -> Option<&str>) = match arch {
+        Arch::X86 => {
+            if obj.is_64() {
+                (Mode::Mode64, describe_group_x86)
+            } else {
+                (Mode::Mode32, describe_group_x86)
+            }
+        }
+        Arch::ARM64 => (Mode::Arm, describe_group_aarch64),
+        _ => {
+            panic!("Unsupported architecture: {:?}", arch);
+        }
     };
 
+    // Disassemble the file
     let mut cs =
-        Capstone::new_raw(Arch::X86, mode, NO_EXTRA_MODE, None).expect("can't initialize capstone");
+        Capstone::new_raw(arch, mode, NO_EXTRA_MODE, None).expect("can't initialize capstone");
```

Comment:
So, can we iterate over architectures here and look at whether the result is OK or Err, instead of calling `unwrap()` directly? And then, if we've exhausted our list and still not found success, then we error out of the program.

Reply:
Unfortunately Capstone is happy to interpret a binary in the wrong architecture, it either doesn't read or check the headers. So running with --x86 on an Arm binary doesn't fail, it just gives an odd answer. There might be a way to look for invalid instructions, but I don't know if that could be done reliably or if data in text segments would cause problems.

Reply:
Ahhh. I feel like there's got to be a way to quickly peek into the headers of the executable to figure out its architecture, but maybe that's not as easy as it "should" be.

Let me take a few days and see if I can find the chance to sit down and take a look at whether we can make this automagical. If not, I'll go ahead and review as-is — some support is better than none, and this tool isn't important enough to hold up a useful contribution in the name of architectural purity.

Thanks for taking the time to look into this and explain it to me!

Reply:
I think the automagic is very easy! In the `main()` function, it looks to me like `obj.architecture()` (from `object::File::parse()`) will return an value that will tell us the architecture of the binary empirically.

I'm happy to update this PR to include functionality to use this information, but based on your comment in the associated issue, it sounded like maybe your code isn't quite settled yet.